### PR TITLE
[Hub Generated] Review request for Microsoft.Devices to add version preview/2022-08-07

### PR DESCRIPTION
**Pull request by Selekena**
This is a PR generated at [OpenAPI Hub](https://aka.ms/openapihub). You can view your [work branch via this link](https://portal.azure-devex-tools.com/branch/Selekena/azure-rest-api-specs/dev-deviceprovisioningservices-Microsoft.Devices-2022-08-07...main/).
## ARM API Information (Control Plane)

Azure 1st Party Service can try out the [Shift Left](https://aka.ms/ShiftLeft) experience to initiate API design review from ADO code repo.  If you are interested, may request engineering support by filling in with the form https://aka.ms/ShiftLeftSupportForm.

### Changelog
Add a changelog entry for this PR by answering the following questions:
  1. What's the purpose of the update?
      - [ ] new service onboarding
      - [x] new API version
      - [ ] update existing version for new feature
      - [ ] update existing version to fix swagger quality issue in s360
      - [ ] Other, please clarify
  2. When are you targeting to deploy the new service/feature to public regions? Please provide the date or, if the date is not yet available, the month.
  3. When do you expect to publish the swagger? Please provide date or, the the date is not yet available, the month.
  4. By default, Azure SDKs of all languages (.NET/Python/Java/JavaScript for both management-plane SDK and data-plane SDK, Go for management-plane SDK only ) MUST be refreshed with/after swagger of new version is published. If you prefer NOT to refresh any specific SDK language upon swagger updates in the current PR, please leave details with justification here.

### Contribution checklist (MS Employees Only):
- [x] I commit to follow the [Breaking Change Policy](https://aka.ms/AzBreakingChangesPolicy) of "no breaking changes"
- [x] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [ ] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist

> **Applicability**: :warning:
>
> If your changes encompass only the following scenarios, you should SKIP this section, as these scenarios do not require ARM review.
> - Change to data plane APIs
> - Adding new properties
> - All removals

Otherwise your PR may be subject to ARM review requirements. Complete the following:
- [x] Check this box if any of the following apply to the PR so that the label "ARMReview" and "WaitForARMFeedback" will be added by bot to kick off ARM API Review.  Missing to check this box in the following scenario may result in delays to the ARM manifest review and deployment.
  - Adding a new service
  - Adding new API(s)
  - Adding a new API version
    -[ ] To review changes efficiently, ensure you copy the existing version into the new directory structure for first commit and then push new changes, including version updates, in separate commits. You can use OpenAPIHub to initialize the PR for adding a new version. For more details refer to the [wiki](https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/208/OpenAPI-Hub-Adding-new-API-version).

- [x] Ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [ ] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them.

### Breaking Change Review Checklist
If you have any breaking changes as defined in the [Breaking Change Policy](https://aka.ms/AzBreakingChangesPolicy/), request approval from the Breaking Change Review Board.

**Action**: to initiate an evaluation of the breaking change, create a new intake using the [template for breaking changes](https://aka.ms/Breakingchangetemplate). Additional details on the process and office hours are on the [Breaking Change Wiki](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37684/Breaking-Changes).

NOTE: To update API(s) in public preview for over 1 year (refer to [Retirement of Previews](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37683/Retirement-of-Previews))

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

File: specification/deviceprovisioningservices/resource-manager/Microsoft.Devices/preview/2022-08-07/iotdps.json [hunk 1]
```diff
@@ -0,0 +1,2602 @@
...
+    "Operation": {
+      "description": "Provisioning Service REST API operation.",
+      "type": "object",
+      "properties": {
+        "name": {
+          "readOnly": true,
+          "type": "string",
+          "description": "Operation name: {provider}/{resource}/{read | write | action | delete}"
+        },
+        "isDataAction": {
```
Comment (jorgecotillo) [inline specification/deviceprovisioningservices/resource-manager/Microsoft.Devices/preview/2022-08-07/iotdps.json:2299]: >isDataAction [](http://example.com/codeflow?start=9&length=12)

Did you miss: `"readOnly": true` mark this property as readonly if your RP is setting it.

**Comment by raych1** [comment]
@Selekena , can you fix the CI check errors?

File: specification/deviceprovisioningservices/resource-manager/readme.md [hunk 2]
```diff
@@ -26,19 +26,28 @@ These are the global settings for the API.
...
+```yaml $(tag) == 'package-preview-2022-08'
+input-file:
+  - Microsoft.Devices/preview/2022-08-07-preview/iotdps.json
+```
 ### Tag: package-2022-02
 
 These settings apply only when `--tag=package-2022-02` is specified on the command line.
 
-```yaml $(tag) == 'package-2022-02'
+``` yaml $(tag) == 'package-2022-02'
```
Comment (mikekistler) [inline specification/deviceprovisioningservices/resource-manager/readme.md:46]: I recommend that you revert the change on this line. Maybe this will fix the whacky PR checks.
```suggestion
```yaml $(tag) == 'package-2022-02'
```

**Comment by raych1** [comment]
@dw511214992 , can you have a look at SDK automation task failures? It doesn't seem cause by PR change.

**Comment by raych1** [comment]
@msyyc , can you have a look at the python SDK failures?

**Comment by raych1** [comment]
@Selekena , looks like this version hasn't been deployed on server side? When will that happen?

**Comment by Selekena** [comment]
This branch needs to be abandoned in favor of this work stream: https://github.com/Azure/azure-rest-api-specs-pr/pull/8519.
